### ISSUE TITLE
F1

### ISSUE THREAD

**Issue by gunnxx** [yidingjiang/love#1]
Hi,

I am wondering whether you have the script to compute directly the results of Table 2 ie. precision, recall, and F1.

Thanks.

**Comment by ezliu** [comment]
The results in Table 2 can be obtained by running `train_rl.py` with the appropriate arguments. For example, you can run LOVE with the following command from the `README` (see `README` for how to appropriately add wandb):

```
PYOPENGL_PLATFORM=egl python train_rl.py \
    --name=grid_world_love \
    --coding_len_coeff=0.005 \
    --kl_coeff=0.0 \
    --rec_coeff=1.0 \
    --use_abs_pos_kl=1.0 \
    --batch-size=64 \
    --dataset-path=compile  \
    --max-iters=20000  \
    --state-size=8 \
    --use_min_length_boundary_mask \
    --latent-n=10 \
    --wandb
```

Then, the precision / recall / F1 numbers can be grabbed from the wandb.

**Comment by gunnxx** [comment]
I have one additional question. Are there any scripts to generate the dataset? I am curious about how you generated the dataset. How can you make all the episode length becomes 18?

**Comment by ezliu** [comment]
Unfortunately, I'm not sure where our script to generate the dataset is. However, the already-generated dataset is available (see the README). We generated the dataset by computing the shortest path that picks up all of the objects in the correct order. We chose to use demonstrations in the dataset of all the same length for simplicity, so we would not have to implement padding (to handle batches of variable length episodes). This was achieved by creating demonstrations for random environments and then filtering out all demonstrations except for those with the most common length: i.e., 18. However, in principle, there is nothing in the LOVE algorithm that depends on the dataset containing episodes of the same length, and variable length episodes can be handled by implementing padding inside `hssm_rl.py`. Additionally, notably the test time tasks are not filtered in any way, and in particular, many of the test episodes have a different length (i.e., not 18), suggesting that the skills learned with LOVE generalize to different lengths.
